### ISSUE TITLE
Cross-device link fail in truffle init

### ISSUE THREAD

**Issue by nolash** [trufflesuite/truffle#3181]
- [ ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

I dont use gitter

---------------------------

## Issue

truffle init fails when tmp folder on different device

## Steps to Reproduce

truffle init

## Expected Behavior

it to work.

## Actual Results

```
decentralized-resolver $ truffle init
This directory is non-empty...
? Proceed anyway? (Y/n) 
Starting unbox...
=================

? Proceed anyway? Yes
✔ Preparing to download box
⠙ DownloadingError: EXDEV: cross-device link not permitted, rename '/home/lash/src/home/decentralized-resolver/1595096743958-8729593474794697/bare-box-master' -> '/tmp/tmp-19390-ugu3236Ok7ZD/box'
```

## Environment

* Operating System: linux 5.7.9
* Ethereum client: n/a
* Truffle version (`truffle version`): 5.1.35
* node version (`node --version`): 14.5.0, 10.16.0
* npm version (`npm --version`): 6.14.6, 6.9.0

**Comment by nolash** [comment]
Actually it seems to happen even if it's not cross-device

**Comment by CruzMolina** [comment]
Hey @nolash ! Can you provide exact reproduction steps?

I tried to duplicate using `mktemp -d -t test-XXXXXXXXXX`, `cd`ing into the generated directory and running `truffle init` using the latest version of `truffle` globally installed, but the command worked as expected.

**Comment by nolash** [comment]
@CruzMolina I don't quite know what else to include:

```
npm install
alias truffle=<pwd>/node_modules/truffle/build/<that_cli_file.js>
truffle init
```

**Comment by t4sk** [comment]
This worked for me

```shell
TEMP=$PWD truffle init
```

More info here https://github.com/trufflesuite/truffle/issues/351

**Comment by nolash** [comment]
@t4sk thanks although I am not using vm

**Comment by nolash** [comment]
@t4sk `TEMP` env var worked for me, thanks.

**Comment by CruzMolina** [comment]
@nolash does the error persist when using `truffle` v5.1.34?

**Comment by nolash** [comment]
actually it does unfortunately, _but_ it turns out I _do_ have a different partition for `/tmp` - sorry :/

```
tmpfs on /tmp type tmpfs (rw,nosuid,nodev)
```

**Comment by nolash** [comment]
well not different partition, but this builtin thing.

So this is an issue for generic `archlinux` in general, I would guess

**Comment by haltman-at** [comment]
Note: I'm also getting basically this same error when I try to run the debugger tests.  I'm on Linux Mint 20.  Will need to try reproducing this in the reported situation to make it directly analogous, but just wanted to get that noted here first.

**Comment by haltman-at** [comment]
Yup, here's what I get when I run `truffle init`:
```
sniffnoy@hyperpotato ~/truffle/bare $ truffle init

Starting unbox...
=================

✔ Preparing to download box
⠋ DownloadingError: EXDEV: cross-device link not permitted, rename '/home/sniffnoy/truffle/bare/1595358244292-18816652320277494/bare-box-master' -> '/tmp/tmp-22331-8N7FeaPuxExe/box'
```

**Comment by haltman-at** [comment]
OK, on investigating this, the problem seems to be coming from the invocation of `github-downloader` on line 42 of this file: https://github.com/trufflesuite/truffle/blob/develop/packages/box/lib/utils/unbox.ts

**Comment by damien-lloyd** [comment]
I'm getting this problem as well, but I don't have a separate /tmp partition. `mktemp -d -t test-XXXXXXXXXX` works on my machine, but `truffle init` and `truffle unbox` fail with 

```
EXDEV: cross-device link not permitted, rename '/home/damien/Projects/election/1595425269981-5694624604979723/bare-box-master' -> '/tmp/tmp-183015-I6511iEZEQpt/box'
```

Running `TEMP=$PWD truffle init` makes the command run successfully.

OS: Manjaro 64-bit

**Comment by gnidan** [comment]
This is in progress. Related PR: #2987

**Comment by eggplantzzz** [comment]
This should be fixed on version 5.1.36
